File: imhere-notification-service/src/main/java/com/kdongsu5509/imherenotificationservice/adapter/in/FcmTokenEnrollController.java
package com.kdongsu5509.imherenotificationservice.adapter.in;

import com.kdongsu5509.imherenotificationservice.adapter.in.dto.FcmTokenInfo;
import lombok.extern.slf4j.Slf4j;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController("/api/v1/noti")
public class FcmTokenEnrollController {

    @PostMapping("/enroll")
    void enroll(@Validated @RequestBody FcmTokenInfo fcmTokenInfo) {

    }
}

File: src/main/kotlin/com/kdongsu5509/notifications/adapter/in/FcmTokenEnrollController.kt
package com.kdongsu5509.notifications.adapter.`in`

import com.kdongsu5509.notifications.adapter.`in`.dto.FcmTokenInfo
import com.kdongsu5509.notifications.application.port.`in`.SaveFcmTokenUseCasePort
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/notification")
class FcmTokenEnrollController(
    private val saveFcmTokenUserCasePort: SaveFcmTokenUseCasePort
) {
    @PostMapping("/enroll")
    fun enroll(
        @Validated @RequestBody fcmTokenInfo: FcmTokenInfo,
        @AuthenticationPrincipal userDetails: UserDetails
    ) {
        val userEmail = userDetails.username
        saveFcmTokenUserCasePort.save(fcmTokenInfo.fcmToken, userEmail, fcmTokenInfo.deviceType)
    }
}

